infra/modules/providers/azure/app-service/main.tf

```diff
@@ -0,0 +1,13 @@
+resource "azurerm_resource_group" "appsvc" {
+  name     = "${var.resource_group_name}"
+  location = "${var.resource_group_location}"
+  tags     = "${var.resource_tags}"
+}
+
+resource "azurerm_app_service" "appsvc" {
+  name                = "${var.app_service_name}"
+  location            = "${azurerm_resource_group.appsvc.location}"
+  resource_group_name = "${azurerm_resource_group.appsvc.name}"
```

Comment:
to be clear, the resource group reference should be the one from the app service plan.

Reply:
The referencing happens at azure-simple level as compared to module definition level as explained above.

Reply:
the variable name is`azurerm_resource_group.appsvc.name` which is misleading as the resource group name is sourced from the app service plan.

Reply:
There are a couple of reasons for this approach:
1. The module app service will be consistent on the resource names.
2. While debugging an issue, module reference name will always help since every module will have a reference name.
3. If there are multiple resources referenced from previous modules, tracking back to where the resource came from will be easier for the user during deployment.